src/store/epics/transfer.ts

```diff
@@ -1,25 +1,57 @@
-import { of } from 'rxjs'
+import { merge, of } from 'rxjs'
 import { fromPromise } from 'rxjs/observable/fromPromise'
-import {
-  catchError,
-  filter,
-  flatMap,
-  ignoreElements,
-  map,
-  mergeMap,
-  withLatestFrom,
-} from 'rxjs/operators'
+import { catchError, filter, map, mapTo, mergeMap, withLatestFrom } from 'rxjs/operators'
 import { isActionOf } from 'typesafe-actions'
 
 import {
-  accountLoadRequest,
+  insufficientFunds,
+  publicKeyValidation,
+  showNotification,
   transactionFailed,
   transactionRequest,
   transactionSuccess,
   transferRequest,
+  updateContactForm,
+  updateFee,
+  updateRemainingBalance,
+  updateTransferForm,
 } from '@actions'
 import { getActiveAccount } from '@selectors'
+import { getFeeInKinesis } from '@services/kinesis'
 import { RootEpic } from '@store'
+import { NotificationType } from '@types'
+
+export const amountCalculations$: RootEpic = (action$, state$, { getCurrentConnection }) =>
+  action$.pipe(
+    filter(isActionOf(updateTransferForm)),
+    filter(({ payload: { field } }) => field === 'amount'),
+    filter(
+      ({ payload: { newValue } }) => newValue === '' || /^[0-9]+(\.)?([0-9]{1,5})?$/.test(newValue),
+    ),
+    withLatestFrom(state$),
+    mergeMap(([action, state]) => {
```

Comment:
I feel like this could be a time when we don't want to use `mergeMap` because it can result in multiple simultaneous requests which may not return in the order we anticipate.

Something like `exhaustMap` might be what we want instead, especially if we are going to make calculations based on values returned from the promise/request we are making. The difference is that exhaustMap will wait until the current request has completed/errored before taking on the next one.

Reply:
exhaustMap is in